File: leetcode/js/pascals-triangle-ii.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function (rowIndex) {
  const result = Array(rowIndex + 1).fill(1);

  for (let i = 1; i < rowIndex; i++) {
    for (let j = i; j > 0; j--) {
      result[j] += result[j - 1];
    }
  }

  return result;
};

var getRow = function (rowIndex) {
  let row = [1];

  for (let i = 0; i < rowIndex; i++) {
    row.push((row[i] * (rowIndex - i)) / (i + 1));
  }

  return row;
};

File: leetcode/rust/pascals-triangle-ii.rs
impl Solution {
    pub fn get_row(row_index: i32) -> Vec<i32> {
        let mut result = vec![1; (row_index + 1) as usize];

        for i in 1..row_index {
            for j in (1..=i).rev() {
                result[j as usize] += result[(j - 1) as usize];
            }
        }

        result
    }
}
